tests/acct_mgt_test.py

```diff
@@ -0,0 +1,353 @@
+    It is used to wait for openshift to do something with the pod, as in, deleting
+    or starting a pod.
+
+    pass in the following parameter:
+       project:  the namespace of the pod
+       pod_name: the name of the pod
+       statuses: array of statuses
+
+
+    """
+    time_left = time_out
+    time_interval = 5
+    time.sleep(time_interval)
+    status = get_pod_status(project, pod_name)
+    while status in statuses and time_left > 0:
+        time.sleep(time_interval)
+        time_left = time_left - time_interval
+        status = get_pod_status(project, pod_name)
+
+    if status in statuses:
+        return False
+    return True
+
+
+def wait_until_done(oc_cmd, finished_pattern, time_out=30, decrement=5):
+    """
+    This wait while an oc command is running, looking for a pattern that
+    indicates it is finished.
+    """
+    pattern = re.compile(finished_pattern)
+    done = False
+    oc_array = oc_cmd.split(" ")
+    matched_line = ""
+    while time_out > 0 and not done:
+        time.sleep(5)
+        time_out = time_out - decrement
+        result = subprocess.run(
+            oc_array,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+        line_list = result.stdout.decode("utf-8").split("\n")
+        for line in line_list:
+            if pattern.match(line):
+                matched_line = line
+                done = True
+    if pattern.match(matched_line):
+        return True
+    return False
+
+
+def compare_results(result, pattern):
+    """This compares the result of a subprocess (usually curl) call  with a pattern"""
+    if result is not None:
+        pattern = re.compile(pattern)
+        line_array = result.stdout.decode("utf-8").split("\n")
+        for line in line_array:
+            if pattern.match(line):
+                return True
+    return False
+
+
+def oc_resource_exist(resource, kind, name, project=None):
+    """This uses oc to determine if an openshift resource exists"""
+    result = None
+    if project is None:
+        result = subprocess.run(
+            ["oc", "-o", "json", "get", resource, name],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+    else:
+        result = subprocess.run(
+            ["oc", "-o", "json", "-n", project, "get", resource, name],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+    if result.returncode == 0:
+        if result.stdout is not None:
+            result_json = json.loads(result.stdout.decode("utf-8"))
+            if result_json["kind"] == kind and result_json["metadata"]["name"] == name:
+                return True
+    return False
+
+
+def ms_check_project(acct_mgt_url, project_name, auth_opts=None):
+    """Checks if the project exists using the microserver"""
+    cmd = (
+        ["curl", "-X", "GET", "-kv"]
```

Comment:
sorry it's not clear to me why we are using curl and subprocess instead of using python's requests library here.

Reply:
I could use python's request library for what I am doing with curl.

subprocess is being used to run 'oc'

since I needed to run oc I just went with curl as opposed to using subprocess and requests.

That is a bit larger change than I want to add during this PR.

Reply:
it's okay for this PR, but I think at some point we should replace the usage of curl with `requests`.

Reply:
I am currently working on replacing curl with requests.